test/Worker.Extensions.Tests/Cosmos/CosmosDBConverterTests.cs

```diff
@@ -359,14 +360,18 @@ public async Task ConvertAsync_CosmosContainerIsNull_ThrowsException_ReturnsFail
             Assert.Equal($"Unable to create Cosmos container client for 'myContainer'.", conversionResult.Error.Message);
         }
 
-        [Fact]
-        public async Task ConvertAsync_POCO_IdProvided_StatusNot200_ThrowsException_ReturnsFailure()
+        [Theory]
```

Comment:
Can you also add an E2E test to validate this use case?

- Test can go in [CosmosDBEndToEndTests](https://github.com/Azure/azure-functions-dotnet-worker/blob/main/test/E2ETests/E2ETests/Cosmos/CosmosDBEndToEndTests.cs)
- If needed, you can create a new cosmos function [here](https://github.com/Azure/azure-functions-dotnet-worker/blob/main/test/E2ETests/E2EApps/E2EApp/Cosmos/CosmosFunction.cs), but given it's an 404 test, you can probably use an existing function